types/Services/baseResponse.d.ts

```diff
@@ -0,0 +1,6 @@
+export type BaseResponse<T> = {
+    success: boolean;
+    message: string;
+    next?: number;
+    data: T;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `BaseResponse` type definition looks good. It's a common practice to have a standard response format for APIs, and this type seems to cover the essential fields: `success`, `message`, `next`, and `data`. However, it would be better if you could add some comments to describe what each field is used for, especially the `next` field which might not be immediately clear to other developers.
```diff
+ /**
+  * Type for API responses
+  * @template T The type of data in the response
+  * @property {boolean} success - Indicates whether the request was successful
+  * @property {string} message - A message describing the result of the request
+  * @property {number} [next] - The next page number for paginated results
+  * @property {T} data - The data returned by the request
+  */
  export type BaseResponse<T> = {
      success: boolean;
      message: string;
      next?: number;
      data: T;
  };
```
Please note that adding comments to your types and interfaces can greatly improve code readability and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->